dev-scripts/dependencies/check-deps.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+#
+# Wrapper script around bazel-deps to generate third_party files, and check if they differ from what's at HEAD.  Exits 0
+# if they match, non-zero otherwise.
+
+set -eu
+
+SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+hash_of_third_party_files() {
+    find thirdparty -type f | sort | xargs sha1sum
+}
+
+BEFORE=$(hash_of_third_party_files)
+
+# shellcheck disable=SC1090
+"$SCRIPT_DIR/generate.sh"
+
+AFTER=$(hash_of_third_party_files)
+
+# Fail if the generated files don't match what's committed.
+if ! diff -u <(echo "$BEFORE") <(echo "$AFTER")
+then
+    echo
+    echo "* * *"
+    echo "Generated dependency files are not up to date.  Run \`dev-scripts/dependencies/generate.sh\` and commit the changes."
```

Comment:
is this something that you found useful in general with `bazel-deps`?
If so, let's open an Issue in `bazel-deps` and see what Oscar thinks.
If he's open to the idea, let's put a # TODO here to delete the file once the upstream issue is resolved?

Reply:
Yes, its occasionally useful. We have `check-deps.sh` run in our CI so that someone doesn't update `maven_deps.yaml` without running the generation step and checking in the `thirdparty` BUILD files. It's saved us a couple of times so far.

Are you thinking something like `"${BAZEL_DEPS_DIR}/gen_maven_deps.sh check -r ${SCRIPT_DIR} -s thirdparty/workspace.bzl -d maven_deps.yaml`?

Reply:
yep, something like that.